drivers/ethernet/eth_adin2111.c

```diff
@@ -805,6 +942,25 @@ static int adin2111_init(const struct device *dev)
 		return ret;
 	}
 
+	if (IS_ENABLED(CONFIG_ETH_ADIN2111_SPI_OA)) {
+		struct gpio_dt_spec spec = {0, 0, 0};
+
+		/*
+		 * OA requires sw chip select control
+		 */
+		ctx->cs_gpio = cfg->spi.config.cs.gpio;
+		cfg->spi.config.cs.gpio = spec;
```

Comment:
That is a bit odd as driver config is supposed to be immutable.

It would be better to solve during config definition with macros: `COND_CODE_1` and `DT_INST_NODE_HAS_PROP`.

When OA property is present, then `spi.config.cs.gpio` can be set to `{0, 0, 0}`, otherwise `SPI_DT_SPEC_INST_GET` is used.

That will require OA to be a DT property as proposed in https://github.com/zephyrproject-rtos/zephyr/pull/68532/files#r1482973313.

Reply:
Thanks for the feedback, will do as suggestd.